lua/autorun/server/sv_focus.lua

```diff
@@ -0,0 +1,19 @@
+local tabbedOutPlys = {}
+
+util.AddNetworkString( "CFC_AttentionMonitor_gameHasFocus" )
+util.AddNetworkString( "CFC_AttentionMonitor_sendData" )
+
+function gameHasFocusCallback( _, pl )
+	local hasFocus = net.ReadBool()
+	if not hasFocus and not tabbedOutPlys[ pl ] then -- Checks to see if it needs to add the player
+		tabbedOutPlys[ pl ] = true
+	else
+		tabbedOutPlys[ pl ] = nil
+	end
+
+	net.Start( "CFC_AttentionMonitor_sendData" ) -- Sends the list of players to the client

```

Comment:
sending a net message to all the players online when the server receives one is highly exploitable like this. if a player spams `CFC_AttentionMonitor_gameHasFocus` to the server the server will spam that exact amount back to everyone online and heavily mess their game up. I'd recommend sending it out with a timer ~1-5 seconds.